### ISSUE TITLE
Use the callsite span of a macro expansion in compiletest

### ISSUE THREAD

**Issue by fpoli** [rust-lang/rust#51848]
The following test fails, because `compiletest` does not look for the callsite span of a macro expansion:

```
macro_rules! macro_with_error {
    ( ) => {
        let x: u64 = true;
    };
}

fn main() {
    macro_with_error!();  //~ ERROR mismatched types
}
```

This causes some problems when the macro definition is not in the file that is tested, see issue https://github.com/laumann/compiletest-rs/issues/48, because there is no way to expect the error with an `//~ ERROR` comment.

Does the current implementation ignore the callsite of macro expansions on purpose?
If not, I have a fix for this: https://github.com/laumann/compiletest-rs/pull/116

**Comment by Mark-Simulacrum** [comment]
cc @nrc @jseyfried 

It might be best to open a PR with the changes if that's relatively trivial to do; I think you'd be more likely to get faster feedback then.

**Comment by fpoli** [comment]
Fixed in 6a1c0637ce44aeea6c60527f4c0e7fb33f2bcd0d
